cos-fleetshard-operator-debezium/src/main/java/org/bf2/cos/fleetshard/operator/debezium/model/ConnectorStatus.java

```diff
@@ -22,12 +26,50 @@ public ConnectorStatusSpec getStatusSpec() {
     }
 
     public List<Condition> getStatusSpecConditions() {
+        return new ArrayList<>(getConditions().values());
+    }
+
+    private Map<String, Condition> getConditions() {
         if (null == statusSpecConditions) {
-            statusSpecConditions = statusSpec.getConditions();
+            statusSpecConditions = new HashMap<>(statusSpec.getConditions().size());
+            for (var condition : statusSpec.getConditions()) {
+                statusSpecConditions.put(condition.getType(), condition);
+            }
         }
         return statusSpecConditions;
     }
 
+    private boolean isSimilarCondition(Condition condition, Condition otherCondition) {
+        if (condition == null && otherCondition != null) {
+            return false;
+        }
+
+        assert condition != null;
+        if (!Objects.equals(condition.getType(), otherCondition.getType())) {
+            return false;
+        } else if (!Objects.equals(condition.getStatus(), otherCondition.getStatus())) {
+            return false;
+        } else if (!Objects.equals(condition.getReason(), otherCondition.getReason())) {
+            return false;
+        } else {
+            return Objects.equals(condition.getMessage(), otherCondition.getMessage());
+        }
+    }
+
+    private boolean hasSimilarCondition(Condition condition) {
+        var lastCondition = getConditions().get(condition.getType());
+        if (null == lastCondition) {
+            return false;
+        }
+        return isSimilarCondition(lastCondition, condition);
+    }
+
+    public void setCondition(Condition condition) {
+        if (!hasSimilarCondition(condition)) {
+            getConditions().put(condition.getType(), condition);
```

Comment:
Only add or replace when the condition content is new. Else we keep the previous condition/s.